src/api/api.ts

```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+
+const BASE_URL = "https://bootcamp-api.codeit.kr";
+
+async function getAPI(query: string) {
+  const response = await axios
+    .get(`${BASE_URL}${query}`)
+    .then((res) => res.data);
+
+  return response;
+}
+
+export const sampleUserData = () => {
+  return getAPI("/api/sample/user");
+};
```

Comment:
P2:
axios로 instance를 생성하고 아래처럼 작성해도 동일동작합니다.

```suggestion
const BASE_URL = "https://bootcamp-api.codeit.kr/api";
const axiosInstance = axios.create({
  baseURL: BASE_URL
})

async function getAPI(query: string) {
  const response = await axiosInstance
    .get(query)
    .then((res) => res.data);

  return response;
}

export const sampleUserData = () => {
  return getAPI("/sample/user");
};
```

Reply:
axios로 instance를 생성하는게 어떤부분이 좋은건가요??

Reply:
axios 인스턴스는 주어진 옵션으로 인스턴스를 생성해, baseUrl, timeout, header와 같이 요청별로 반복되는 코드를 줄일 수 있게 해줍니다. 지금 경미님 코드에서 getAPI라는 함수를 통해 baseURL을 입력하지 않아도 되게 작업해주셨지만, 이와 비슷한 역할을 axios 인스턴스가 해준다고 생각해주시면 좋을 것 같습니다.

또한 나중에 post 요청을 보내야 할 때가 생기게 되면 postAPI와 같이 query만을 받아 post요청을 날리는 함수를 생성하셔야 하는데, 이 요청시 사용되는 설정 값이 post, get 요청이 모두 동일할 가능성이 크니, 인스턴스를 생성해 코드 반복을 줄이며 가독성에 도움이 된다고 생각해 axios instance 생성을 추천드렸습니다~